server/checks/Approval.js

```diff
@@ -148,66 +161,111 @@ export default class Approval extends Check {
     return promiseReduce(comments, checkComment, {total: 0, groups: {}})
   }
 
+  static async getVetosForConfig(github, repository, comments, config, token) {
+    const that = this
+
+    async function checkComment(counter, comment) {
+      if (config.from) {
+        const matches = await that.doesCommentMatchConfig(github, repository, comment, config.from, token)
+        if (matches) {
+          info(`${repository.full_name}: Counting ${comment.user.login}'s veto`)
+          return counter + 1
+        }
+        else {
+            return counter
+        }
+      } else {
+        // if there is no from clause, every veto counts
+        return counter + 1
+      }
+    }
+
+    return promiseReduce(comments, checkComment, 0)
+  }
+
   /**
-   * Removes comments from ignored users, comments that do not match approval
-   * pattern as well as multiple approvals by the same person and counts the
-   * remaining approvals.
+   * Removes comments from ignored users for approvals, comments that do not match approval/veto
+   * pattern as well as multiple approvals/vetos by the same person and counts the
+   * remaining approvals/vetos.
    *
    * @param github The GithubService instance
    * @param repository The repository
    * @param comments The comments to process
    * @param config The approval configuration
    * @param token The access token to use
-   * @returns {Object} Object of the shape {total: int, groups: { groupName: int } }
+   * @returns {Object} Object of the shape {approvals: {total: int, groups: { groupName: int }}, vetos: int }
    */
-  static async countApprovals(github, repository, comments, config, token) {
-    const {pattern, ignore} = config
+  static async countApprovalsAndVetos(github, repository, comments, config, token) {
+    const ignore = config.ignore
+    const approvalPattern = config.pattern
+    const vetoPattern = _.get(config, 'veto.pattern')
+
     const fullName = `${repository.full_name}`
+    // slightly unperformant filtering here:
+    const containsAlreadyCommentByUser = (c1, i, cmts) => i === cmts.findIndex(c2 => c1.user.login === c2.user.login)
```

Comment:
Since we're not relying on any closures here we can move it out of this function. Maybe rename to `filterUniqueComments` or something?